wren-tests/src/test/java/io/wren/testing/snowflake/AbstractWireProtocolTestWithSnowflake.java

```diff
@@ -0,0 +1,114 @@
+import static io.wren.base.config.SnowflakeConfig.SNOWFLAKE_JDBC_URL;
+import static io.wren.base.config.SnowflakeConfig.SNOWFLAKE_PASSWORD;
+import static io.wren.base.config.SnowflakeConfig.SNOWFLAKE_SCHEMA;
+import static io.wren.base.config.SnowflakeConfig.SNOWFLAKE_USER;
+import static io.wren.base.config.WrenConfig.WREN_DATASOURCE_TYPE;
+import static io.wren.main.sqlglot.SQLGlot.Dialect.SNOWFLAKE;
+import static java.lang.System.getenv;
+import static java.util.Objects.requireNonNull;
+
+public abstract class AbstractWireProtocolTestWithSnowflake
+        extends AbstractWireProtocolTest
+{
+    protected TestingSQLGlotServer testingSQLGlotServer;
+
+    protected static final Manifest DEFAULT_MANIFEST = Manifest.builder()
+            .setCatalog("canner-cml")
+            .setSchema("tpch_tiny")
+            .build();
+
+    @Override
+    protected TestingWrenServer createWrenServer()
+            throws Exception
+    {
+        ImmutableMap.Builder<String, String> propBuilder = ImmutableMap.<String, String>builder()
+                .put(WREN_DATASOURCE_TYPE, SNOWFLAKE.name())
+                .put(SNOWFLAKE_JDBC_URL, getenv("SNOWFLAKE_JDBC_URL"))
+                .put(SNOWFLAKE_USER, getenv("SNOWFLAKE_USER"))
+                .put(SNOWFLAKE_PASSWORD, getenv("SNOWFLAKE_PASSWORD"))
+                .put(SNOWFLAKE_DATABASE, "SNOWFLAKE_SAMPLE_DATA")
+                .put(SNOWFLAKE_SCHEMA, "TPCH_SF1")
+                .put(SQLGLOT_PORT, String.valueOf(testingSQLGlotServer.getPort()));
+
+        try {
+            Path dir = Files.createTempDirectory("wren-snowflake-test");
+            if (getWrenMDLPath().isPresent()) {
+                Files.copy(Path.of(getWrenMDLPath().get()), dir.resolve("mdl.json"));
+            }
+            else {
+                Files.write(dir.resolve("manifest.json"), Manifest.MANIFEST_JSON_CODEC.toJsonBytes(DEFAULT_MANIFEST));
+            }
+            propBuilder.put("wren.directory", dir.toString());
+        }
+        catch (Exception ex) {
+            throw new RuntimeException(ex);
+        }
+
+        Map<String, String> properties = new HashMap<>(propBuilder.build());
+        properties.putAll(properties());
+
+        return TestingWrenServer.builder()
+                .setRequiredConfigs(properties)
+                .build();
+    }
+
+    protected Map<String, String> properties()
+    {
+        return ImmutableMap.of();
+    }
+
+    protected TestingSQLGlotServer prepareSQLGlot()
+    {
+        SQLGlotConfig config = createConfigWithFreePort();
+        return new TestingSQLGlotServer(config);
+    }
+
+    protected Optional<String> getWrenMDLPath()
+    {
+        return Optional.of(requireNonNull(getClass().getClassLoader().getResource("snowflake/mdl.json")).getPath());
+    }
+
+    @Override
+    protected String getDefaultCatalog()
+    {
+        return "canner-cml";
+    }
+
+    @Override
+    protected String getDefaultSchema()
+    {
+        return "tpch_tiny";
+    }
```

Comment:
```suggestion
    }
    @Override
    protected void cleanup()
    {
       // clean the generated schema
    }
```
I guess that we will create something like `pg_catalog` in Snowflake. Just like BigQuery, we should clean up it after tested.